lms/djangoapps/course_api/blocks/transformers/block_counts.py

```diff
@@ -0,0 +1,44 @@
+from openedx.core.lib.block_cache.transformer import BlockStructureTransformer
+
+
+class BlockCountsTransformer(BlockStructureTransformer):
+    """
+    ...
+    """
+    VERSION = 1
+    BLOCK_COUNTS = 'block_counts'
+
+    def __init__(self, block_types_to_count):
+        self.block_types_to_count = block_types_to_count
+
+    @classmethod
+    def collect(cls, block_structure):
+        """
+        Collects any information that's necessary to execute this transformer's
+        transform method.
+        """
+        # collect basic xblock fields
+        block_structure.request_xblock_fields('category')
+
+    def transform(self, user_info, block_structure):
+        """
+        Mutates block_structure based on the given user_info.
```

Comment:
Do we do mutation for performance (as opposed to having immutable block_structures)? Or because they're cumbersome to create...?